idaes/generic_models/unit_models/distillation/tray_column.py

```diff
@@ -0,0 +1,311 @@
+        doc="""property parameter object used to define property calculations,
+**default** - useDefault.
+**Valid values:** {
+**useDefault** - use default package from parent model or flowsheet,
+**PropertyParameterObject** - a PropertyParameterBlock object.}"""))
+    CONFIG.declare("property_package_args", ConfigBlock(
+        implicit=True,
+        description="arguments to use for constructing property packages",
+        doc="""a ConfigBlock with arguments to be passed to a property block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}"""))
+
+    def build(self):
+        """Build the model.
+
+        Args:
+            None
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super(TrayColumnData, self).build()
+
+        # Create set for constructing indexed trays
+        self.tray_index = RangeSet(1, self.config.number_of_trays)
+
+        # Add trays
+
+        # TODO:
+        # 1. Add support for multiple feed tray locations
+        # 2. Add support for specifying which trays have side draws
+        self.tray = Tray(self.tray_index,
+                         default={"has_liquid_side_draw":
+                                  self.config.has_liquid_side_draw,
+                                  "has_vapor_side_draw":
+                                  self.config.has_vapor_side_draw,
+                                  "has_heat_transfer":
+                                  self.config.has_heat_transfer,
+                                  "has_pressure_change":
+                                  self.config.has_pressure_change,
+                                  "property_package":
+                                  self.config.property_package,
+                                  "property_package_args":
+                                  self.config.property_package_args},
+                         initialize={self.config.feed_tray_location:
+                                     {"is_feed_tray": True,
+                                      "has_liquid_side_draw":
+                                      self.config.has_liquid_side_draw,
+                                      "has_vapor_side_draw":
+                                      self.config.has_vapor_side_draw,
+                                      "has_heat_transfer":
+                                      self.config.has_heat_transfer,
+                                      "has_pressure_change":
+                                      self.config.has_pressure_change,
+                                      "property_package":
+                                      self.config.property_package,
+                                      "property_package_args":
+                                      self.config.property_package_args}})
+
+        # Add condenser
+        self.condenser = Condenser(
```

Comment:
I assume conditional options for these are future work (i.e. a `has_condenser` option).

Reply:
Yes. For now, the tray column is a simple conventional column with a condenser and reboiler.